frontend/src/common/types.ts

```diff
@@ -19,6 +19,7 @@ import { Id } from './id';
 
 export type FeedVersion = Opaque<number, 'FeedVersion'>;
 export type ChainLabel = Opaque<string, 'ChainLabel'>;
+export type GenesisHash = Opaque<string, 'GenesisHash'>;
```

Comment:
Ahh, this Opaque type is very clever. `Opaque<T, P>` P being phantom data. Really interesting.

Reply:
Yeah, it's poor man's [opaque type aliases from flow](https://flow.org/en/docs/types/opaque-types/), useful since you can't do zero-cost newtypes in JS :)